hooks/post-command

```diff
@@ -25,6 +25,135 @@ plugin_read_list_into_result() {
+    OSSTR="${OS} ${REV}(${ARCH} $(uname -v)"
+    echo ${OSSTR}
+    return
+
+  elif [ "${OS}" = "AIX" ] ; then
+    OSSTR="${OS} $(oslevel) ($(oslevel -r)"
+    echo ${OSSTR}
+    return
+
+  elif [ "${OS}" = "Linux" ] ; then
+    KERNEL=$(uname -r)
+    if [ -f /etc/os-release ] ; then
+      DIST=$(grep '^NAME=' /etc/os-release | sed s/NAME=// | sed s/\"//g)
+      PSUEDONAME=$(grep '^VERSION_ID=' /etc/os-release | sed s/VERSION_ID=// | sed s/\"//g)
+      REV=$(grep '^VERSION=' /etc/os-release | sed s/VERSION=// | sed s/\"//g)
+      OSSTR="${OS} ${DIST} ${REV}(${PSUEDONAME} ${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/alpine-release ] ; then
+      DIST='Alpine'
+      PSUEDONAME=$(sed s/.*\(// < /etc/alpine-release | sed s/\)//)
+      REV=$(sed s/.*release\ // < /etc/alpine-release | sed s/\ .*//)
+      OSSTR="${OS} ${DIST} ${REV}(${PSUEDONAME} ${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/redhat-release ] ; then
+      DIST='RedHat'
+      PSUEDONAME=$(sed s/.*\(// < /etc/redhat-release | sed s/\)//)
+      REV=$(sed s/.*release\ // < /etc/redhat-release | sed s/\ .*//)
+      OSSTR="${OS} ${DIST} ${REV}(${PSUEDONAME} ${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/SuSE-release ] ; then
+      DIST=$(tr "\n" ' ' < /etc/SuSE-release | sed s/VERSION.*//)
+      REV=$(tr "\n" ' ' < /etc/SuSE-release| sed s/.*=\ //)
+      OSSTR="${OS} ${DIST} ${REV}(${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/mandrake-release ] ; then
+      DIST='Mandrake'
+      PSUEDONAME=$(sed s/.*\(// < /etc/mandrake-release | sed s/\)//)
+      REV=$(sed s/.*release\ // < /etc/mandrake-release | sed s/\ .*//)
+      OSSTR="${OS} ${DIST} ${REV}(${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    elif [ -f /etc/debian_version ] ; then
+      if [ "$(awk -F= '/DISTRIB_ID/ {print $2}' /etc/lsb-release)" = "Ubuntu" ]; then
+        DIST="Ubuntu"
+      else
+        DIST="Debian $(cat /etc/debian_version)"
+      fi
+      OSSTR="${OS} ${DIST} (${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    else
+      echo $(uname -a)
+      return
+    fi
+    if [ -f /etc/UnitedLinux-release ] ; then
+      DIST="${DIST}[$(tr "\n" ' ' < /etc/UnitedLinux-release | sed s/VERSION.*//)]"
+      OSSTR="${OS} ${DIST}(${KERNEL} ${MACH})"
+      echo ${OSSTR}
+      return
+    fi
+
+  elif [ "${OS}" == "Darwin" ]; then
+    type -p sw_vers &>/dev/null
+    [ $? -eq 0 ] && {
+      OS=`sw_vers | grep 'ProductName' | cut -f 2`
+      VER=`sw_vers | grep 'ProductVersion' | cut -f 2`
+      BUILD=`sw_vers | grep 'BuildVersion' | cut -f 2`
+      OSSTR="Darwin ${OS} ${VER} ${BUILD}"
+    } || {
+      OSSTR="MacOSX"
+    }
+    echo ${OSSTR}
+    return
+  fi
+
+  echo $(uname -a)
+}
+
+get_codecov_uploader() {
+  local os=$(get_os)
+  local os_array=($os)
+  if [ -n ${os_array[1]} ] && [ ${os_array[1]} = "Alpine" ]; then
+    apk add gnupg
```

Comment:
Does an alpine allow this install on the fly?

Reply:
well, in my test it seems to be working:
```
➜  codecov-buildkite-plugin git:(codecov_uploader) docker run -it alpine:3.14   
/ # gpg
/bin/sh: gpg: not found
/ # apk add gnupg
(1/28) Installing libgpg-error (1.42-r0)
(2/28) Installing libassuan (2.5.5-r0)
(3/28) Installing libcap (2.50-r0)
(4/28) Installing libffi (3.3-r2)
(5/28) Installing libintl (0.21-r0)
(6/28) Installing libblkid (2.37.4-r0)
(7/28) Installing libmount (2.37.4-r0)
(8/28) Installing pcre (8.44-r0)
(9/28) Installing glib (2.68.3-r0)
(10/28) Installing ncurses-terminfo-base (6.2_p20210612-r0)
(11/28) Installing ncurses-libs (6.2_p20210612-r0)
(12/28) Installing libgcrypt (1.9.4-r0)
(13/28) Installing libsecret (0.20.4-r1)
(14/28) Installing pinentry (1.1.1-r0)
Executing pinentry-1.1.1-r0.post-install
(15/28) Installing libbz2 (1.0.8-r1)
(16/28) Installing gmp (6.2.1-r1)
(17/28) Installing nettle (3.7.3-r0)
(18/28) Installing p11-kit (0.23.22-r0)
(19/28) Installing libtasn1 (4.17.0-r0)
(20/28) Installing libunistring (0.9.10-r1)
(21/28) Installing gnutls (3.7.1-r0)
(22/28) Installing libksba (1.5.1-r0)
(23/28) Installing gdbm (1.19-r0)
(24/28) Installing libsasl (2.1.28-r0)
(25/28) Installing libldap (2.4.58-r0)
(26/28) Installing npth (1.6-r0)
(27/28) Installing sqlite-libs (3.35.5-r0)
(28/28) Installing gnupg (2.2.31-r0)
Executing busybox-1.33.1-r7.trigger
OK: 25 MiB in 42 packages
/ # gpg --version
gpg (GnuPG) 2.2.31
libgcrypt 1.9.4
Copyright (C) 2021 Free Software Foundation, Inc.
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /root/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```